end_to_end_tests/golden-record/my_test_api_client/models/a_model.py

```diff
@@ -32,11 +35,14 @@ def to_dict(self) -> Dict[str, Any]:
 
         a_date = self.a_date.isoformat()
 
-        if self.nested_list_of_enums is None:
-            nested_list_of_enums = None
+        required_not_nullable = self.required_not_nullable
+
+        if self.nested_list_of_enums is UNSET:
+            nested_list_of_enums = UNSET
         else:
             nested_list_of_enums = []
             for nested_list_of_enums_item_data in self.nested_list_of_enums:
```

Comment:
@dbanty I'm having a lot of trouble with blocks like this - mypy doesn't like "reassigning"  in the else:
`Incompatible types in assignment (expression has type "List[<nothing>]", variable has type "Unset")`
What I _could_ do is add a `nested_list_of_enums: Union[Unset, List[List[DifferentEnum]]] = UNSET` before the if statement and change the if/else to just be `if self.nested_list_of_enums is not None`

Mypy also complains about line 44 as it thinks the type of `self.nested_list_of_enums` can still be `Unset`. Changing line 40 to be `if isinstance(self.nested_list_of_enums, Unset)` fixes this issue, giving way to a new pattern combining both:

```python
nested_list_of_enums: Union[Unset, List[List[DifferentEnum]]] = UNSET
if not isinstance(self.nested_list_of_enums, Unset):
    nested_list_of_enums = []
    etc.
```

The only problem is this gets pretty icky when trying to implement in the templates + account for stuff that can be `None`. Any advice on how to do this in a way thats less icky to implement?

Reply:
For the if/else thing, you can declare the type before the block like:
```python
nested_list_of_enums: Union[Unset, List[List[DifferentEnum]]]
if self.nested_list_of_enums is UNSET:
    nested_list_of_enums = UNSET
else:
    nested_list_of_enums = []
```
I believe mypy will be fine with that, though I'm not sure it's much better than what you had. Unfortunately I think `is instance` is the only way to handle the branches, because we told mypy it can be any `Unset`, not just `UNSET`. I don't think you can use Literal with an object...

Maybe we could do what Pydantic does and use ellipsis `...` for our unset placeholder?

```python
thing: Union[int, Literal[...]] = ...
if thing is ...:
    ...
else:
   ...
```
I don't know how you or mypy would feel about that. We could also alias `UNSET = Literal[...]` for clearer typing.